das/ipfs_storage_service.go

```diff
@@ -86,6 +105,16 @@ func hashToCid(hash common.Hash) (cid.Cid, error) {
 func (s *IpfsStorageService) GetByHash(ctx context.Context, hash common.Hash) ([]byte, error) {
 	log.Trace("das.IpfsStorageService.GetByHash", "hash", pretty.PrettyHash(hash))
 
+	doPin := false // If true, pin every block related to this batch
+	if s.config.PinAfterGet {
+		if s.config.PinPercentage == 100.0 {
+			doPin = true
+		} else if (rand.Float64() * 100.0) <= s.config.PinPercentage {
```

Comment:
might be good to initialize `rand.Seed`, to avoid risk of getting the same pinning distribution on different nodes

Reply:
Thanks, I had missed that!